src/shims/posix/fs.rs

```diff
@@ -852,7 +888,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
 
         this.assert_target_os("macos", "fstat");
-        this.check_no_isolation("`fstat`")?;
+
+        // Reject if isolation is enabled.
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("`fstat`", reject_with)?;
+            // Set error code as "EBADF" (bad fd)
+            return this.handle_not_found();
+        }
 
         let fd = this.read_scalar(fd_op)?.to_i32()?;
```

Comment:
One thing I noticed in my last PR (and fixed for the existing code): we should make sure that the argument checks (`read_scalar`, `read_pointer`, and so on) are all *before* the isolation check. That way, extremely incorrect arguments (like, using the wrong function prototype, or passing uninit data) are still detected.

This applies to all the functions you touched in this file, I think.

Reply:
It makes sense. I'll move these checks right before the system call ops in shims. Will update the PR soon.